c2corg_ui/static/js/map/baselayerselector.js

```diff
@@ -0,0 +1,263 @@
+goog.provide('app.BaselayerSelectorController');
+goog.provide('app.baselayerSelectorDirective');
+
+goog.require('app');
+goog.require('ngeo.BackgroundLayerMgr');
+goog.require('ol.Attribution');
+goog.require('ol.format.WMTSCapabilities');
+goog.require('ol.layer.Tile');
+goog.require('ol.source.OSM');
+goog.require('ol.source.WMTS');
+goog.require('ol.tilegrid.WMTS');
+
+
+/**
+ * This directive is used to display a map baselayer selector.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.baselayerSelectorDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppBaselayerSelectorController',
+    controllerAs: 'bglayerCtrl',
+    bindToController: true,
+    scope: {
+      'map': '=appBaselayerSelectorMap'
+    },
+    templateUrl: '/static/partials/map/baselayerselector.html'
+  };
+};
+
+
+app.module.directive('appBaselayerSelector', app.baselayerSelectorDirective);
+
+
+/**
+ * @param {angular.$http} $http
+ * @param {ngeo.BackgroundLayerMgr} ngeoBackgroundLayerMgr Background layer
+ *     manager.
+ * @param {string} ignApiKey IGN API key.
+ * @param {app.Alerts} appAlerts
+ * @param {app.Authentication} appAuthentication
+ * @constructor
+ * @export
+ * @struct
+ * @ngInject
+ */
+app.BaselayerSelectorController = function($http, ngeoBackgroundLayerMgr,
+  ignApiKey, appAlerts, appAuthentication) {
+
+  /**
+   * @type {angular.$http}
+   * @private
+   */
+  this.http_ = $http;
+
+  /**
+   * @type {ngeo.BackgroundLayerMgr}
+   * @private
+   */
+  this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr;
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.ignApiKey_ = ignApiKey;
+
+  /**
+   * @type {app.Alerts}
+   * @private
+   */
+  this.alerts_ = appAlerts;
+
+  /**
+   * @type {ol.Map}
+   * @export
+   */
+  this.map;
+
+  /**
+   * @type {Object}
+   * @private
+   */
+  this.cachedLayers_ = {};
+
+  /**
+   * @type {ol.format.WMTSCapabilities}
+   * @private
+   */
+  this.wmtsParser_;
+
+  /**
+   * @type {Object}
```

Comment:
Maybe. Actually I have used the same variable name than in ngeo's example at http://camptocamp.github.io/ngeo/master/examples/backgroundlayerdropdown.js